options.js

```diff
@@ -52,54 +75,42 @@ function createDeleteButton() {
     const deleteButton = document.createElement("button");
     deleteButton.textContent = "Delete";
     deleteButton.className = "DeleteBtn";
-    deleteButton.addEventListener("click", () => {
+    deleteButton.addEventListener("click", async () => {
         const itemToRemove = deleteButton.parentElement;
         itemToRemove.remove();
-        chrome.storage.local.get(
-            { exclusionArray: [] },
-            ({ exclusionArray: oldExclusionArray }) => {
-                const exclusionArray = oldExclusionArray.filter(
-                    (item) => item !== itemToRemove.firstChild.data
-                );
-                chrome.storage.local.set({ exclusionArray });
-            }
-        );
+        const { firstChild: { data: removedExclusion } } = itemToRemove;
+        const { options: { exclusions } } = getLocalStorageKey('options');
+        const newExclusions = exclusions.filter(item => item !== removedExclusion);
+        await updateOptions({ exclusions: newExclusions });
     });
     return deleteButton;
 }
 
-function submitChanges() {
+async function submitChanges() {
     const settingFields = ["moveTabs", "effectWindows", "effectTabGroups"];
-    const [moveTabs, effectWindows, effectTabGroups] =
-        settingFields.map(isPropertyEnabled);
-    chrome.storage.local.set({
-        options: { moveTabs, effectWindows, effectTabGroups },
-    });
+    const [moveTabs, effectWindows, effectTabGroups] = settingFields.map(isOptionEnabled);
+    await updateOptions({ moveTabs, effectWindows, effectTabGroups });
 }
 
-submit.addEventListener("click", () => {
-    submitChanges();
-});
+submit.addEventListener("click", submitChanges);
 
-exclusionBox.addEventListener("keyup", (event) => {
-    if (event.code === "Enter") {
-        if (/\s/.test(exclusionBox.value) || !exclusionBox.value) {
-            return;
-        }
-        chrome.storage.local.get(
-            { exclusionArray: [] },
-            ({ exclusionArray: oldExclusionArray }) => {
-                if (oldExclusionArray.includes(exclusionBox.value)) {
-                    exclusionBox.value = "";
-                    return;
-                }
-                const exclusionArray = [
-                    ...oldExclusionArray,
-                    exclusionBox.value,
-                ];
-                chrome.storage.local.set({ exclusionArray });
-                addToHTMLList(exclusionBox.value);
-            }
-        );
+exclusionBox.addEventListener("keyup", async ({ code }) => {
+    const { value: newExclusion } = exclusionBox;
+    if (
+        code !== 'Enter'
+        || !newExclusion
+        || /\s/.test(newExclusion)
```

Comment:
This change was important. The way lazy evaluation works is you were checking if it was undefined _after_ trying to do something with it. Swap the two and will work as expected.